src/main/java/cc/polyfrost/oneconfig/internal/config/HypixelKeys.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file is part of OneConfig.
+ * OneConfig - Next Generation Config Library for Minecraft: Java Edition
+ * Copyright (C) 2021, 2022 Polyfrost.
+ *   <https://polyfrost.cc> <https://github.com/Polyfrost/>
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ *   OneConfig is licensed under the terms of version 3 of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, AND
+ * under the Additional Terms Applicable to OneConfig, as published by Polyfrost,
+ * either version 1.0 of the Additional Terms, or (at your option) any later
+ * version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ *   You should have received a copy of the GNU Lesser General Public
+ * License.  If not, see <https://www.gnu.org/licenses/>. You should
+ * have also received a copy of the Additional Terms Applicable
+ * to OneConfig, as published by Polyfrost. If not, see
+ * <https://polyfrost.cc/legal/oneconfig/additional-terms>
+ */
+
+package cc.polyfrost.oneconfig.internal.config;
+
+import cc.polyfrost.oneconfig.config.annotations.HypixelKey;
+import cc.polyfrost.oneconfig.config.elements.BasicOption;
+import cc.polyfrost.oneconfig.events.EventManager;
+import cc.polyfrost.oneconfig.events.event.ChatReceiveEvent;
+import cc.polyfrost.oneconfig.events.event.Stage;
+import cc.polyfrost.oneconfig.events.event.TickEvent;
+import cc.polyfrost.oneconfig.libs.eventbus.Subscribe;
+import cc.polyfrost.oneconfig.libs.universal.UChat;
+import cc.polyfrost.oneconfig.platform.Platform;
+import cc.polyfrost.oneconfig.utils.Multithreading;
+import cc.polyfrost.oneconfig.utils.Notifications;
+import cc.polyfrost.oneconfig.utils.TickDelay;
+import cc.polyfrost.oneconfig.utils.hypixel.HypixelUtils;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class HypixelKeys {
+    public static final HypixelKeys INSTANCE = new HypixelKeys();
+    private final List<BasicOption> options = new ArrayList<>();
+    private static final int apiKeyMessageLength = "Your new API key is ".length();
+    private boolean hasSynced = false;
+
+    private HypixelKeys() {
+        EventManager.INSTANCE.register(this);
+    }
+
+    public void addOption(BasicOption option) {
+        options.add(option);
+    }
+
+    @Subscribe
+    private void onChatReceived(ChatReceiveEvent event) {
+        if (Preferences.autoSetHypixelKey) {
+            String message = event.getFullyUnformattedMessage();
+            if (message.startsWith("Your new API key is ")) {
```

Comment:
maybe check the message length before this check. Just saves a little bit of useless processing time

Reply:
*why*